dashboard/src/components/DeploymentForm/DeploymentForm.tsx

```diff
@@ -58,7 +59,13 @@ export default function DeploymentForm() {
   );
 
   useEffect(() => {
-    dispatch(actions.charts.fetchChartVersions(chartCluster, chartNamespace, packageId, pluginObj));
+    dispatch(
+      actions.charts.fetchChartVersions({
+        context: { cluster: chartCluster, namespace: chartNamespace },
+        plugin: pluginObj,
+        identifier: packageId,
+      } as AvailablePackageReference),
```

Comment:
This is what I meant in another comment: in this PR I'm just using the same variables we already had to build the `xxReference`... however, I think we should move to an approach like:

Is this component a view with route params? 
- Yes: merge (with a util function) these params with the ones in the state (if any).
- No, that means it is just a child component used like `<Blablabla props={props}...>`. Rely on the data passed by the parent component.

I can do it here or maybe in a separate PR, I don't mind.